src/FffGcodeWriter.cpp

```diff
@@ -654,6 +654,19 @@ LayerPlan& FffGcodeWriter::processLayer(const SliceDataStorage& storage, int lay
             }
             layer_thickness = getSettingInMicrons("layer_height_0");
         }
+        int64_t z_offset = getSettingInMicrons("z_offset_layer_0");
```

Comment:
Yeah, I see that the original `z` variable is also `int64_t`, so that's a bug in the original code then so nevermind that :)